### ISSUE TITLE
Buffering messages until sinks are configured?

### ISSUE THREAD

**Issue by brian-fsg** [gabime/spdlog#953]
Hi all -

  I'm looking to have several named loggers (eg, 'Network', 'File', etc) that are initialized before any configuration information is read, and that configuration includes what sinks to use.  For part of this, I can use the dist_sink as the sink for each named logger, then add the actual targets (eg, files, stdout, whatever) once a configuration is received.

  However, because of this, I want the loggers to buffer messages until some sort of configuration is read in saying what the sinks will be.  Reading it in and configuring them is easy enough with a boolean state variable ('configured_') and the dist_sink features, and I can add a queue for log_msg objects to be held in until I'm ready to dump to a configured sink.  

  So far, so good, but I thought I'd ask for advice in case I'm missing something tremendously obvious that would be much easier, and, while this works and lets me buffer messages in the loggers until I configure a sink, it *doesn't* let me read in sink *formatting* - eg, I'm queuing post-formatted log messages.  Admittedly my C++ isn't very strong yet and I'm in the weeds with a lot of advanced stuff like templates, parameter packs, etc, and I couldn't figure out how to save a 'queue' of pre-formatted log requests, since the signatures are different depending on the number or arguments.  I think std::function and std::bind can help, but the few not-sure-what-I'm-doing attempts I made ran into issues with the variadic functions for 'log'.

  Any ideas on how to approach that, or is it just more trouble than its worth?  Queued post-formatted messages already covers 98% of what I wanted, but I thought it couldn't hurt to ask.    Worst case I learn something new.

  Thanks, and happy New Year!

**Comment by brian-fsg** [comment]
Thanks - since I'm modifying the source a bit already, I added a copy constructor for log message and then made a std::queue of them - this seems to work, and then has all the other stuff (like level) still included, so I can still filter by level, whereas that'd get destroyed in a queue of string.  But maybe that opens me up to other issues?

I see in mail your original suggestion was to look at the async logger, which I just peeked at now.  Would implementing something that holds off of ANY flush until I explicitly ask it to start flushing be a better approach? 

Thanks again for your advice.
